controllers/topicPage/TopicPage.js

```diff
@@ -42,6 +50,33 @@ class TopicPage {
     }
   }
 
+  async getValidActivity(userId, payload, limit) {
+    let fetch = 0;
+    let results = {
+      noActivity: false,
+      data: []
+    };
+    while (results.data.length < limit && fetch < MAX_FEED_FETCH_LIMIT) {
+      if (fetch !== 0) {
+        payload.offset = parseInt(payload.offset) + parseInt(limit);
+      }
+      let feeds = await this.getFeedTopicPages(payload);
+      if (feeds.noActivity) {
+        results.noActivity = true;
+        break;
+      }
+      let newTopicPagesWithBlock = feeds.data;
+      if (!newTopicPagesWithBlock.length) break;
+
+      for (let post of newTopicPagesWithBlock) {
+        if (post?.is_hide) continue;
+        results.data.push(await this.modifiedFeed(userId, post));
+      }
+      fetch += 1;
+    }
+    return results;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getValidActivity` function fetches valid activity based on a given limit. The logic seems fine, but again, error handling is missing when calling `this.getFeedTopicPages(payload)`. It's important to handle potential errors to prevent unexpected behavior or crashes.

```diff
      let feeds = await this.getFeedTopicPages(payload);
+     try {
+       let feeds = await this.getFeedTopicPages(payload);
+     } catch (error) {
+       console.error('Error fetching feed topic pages:', error);
+       results.noActivity = true;
+       break;
+     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->